pmacApp/src/pmacController.cpp

```diff
@@ -3068,7 +3100,32 @@ asynStatus pmacController::appendToProfile() {
         }
       }
     }
+
+    if (tScanPendingPointReady_ > 0) {
+      memmove(&profileTimes_[1], &profileTimes_[0], (numPointsToBuild) * sizeof(double));
+      // Insert last Time from previous buffer into the first position
+      profileTimes_[0] = tScanPrevBufferTime;
+    }
+
     // Append the points to the store
+    if(tScanPendingPoint_ > 0) {
+      if(tScanPendingPoint_ == tScanAxisMask_) {
+        numPointsToBuild--;
+      } else {
+        debug(DEBUG_ERROR,functionName,"Inconsistent tScanPendingPoint_");
+      }
+    }
+    if(tScanPendingPointReady_ > 0) {
+      if(tScanPendingPointReady_ == tScanAxisMask_) {
+        numPointsToBuild++;
+        tScanPendingPointReady_ = 0;
+      } else {
+        debug(DEBUG_ERROR,functionName, "Inconsistent tScanPendingPointReady_");
+      }
+    }
+
```

Comment:
This section appears to be the only addition from the previous block.  Is is only applicable in this position?